control/control_test.go

```diff
@@ -838,6 +838,32 @@ func TestMetricConfig(t *testing.T) {
 		time.Sleep(100 * time.Millisecond)
 	})
 
+	Convey("config provided by defaults", t, func() {
+		c := New(GetDefaultConfig())
+		c.Start()
+		lpe := newListenToPluginEvent()
+		c.eventManager.RegisterHandler("Control.PluginLoaded", lpe)
+		_, err := load(c, JSONRPCPluginPath)
+		So(err, ShouldBeNil)
+		<-lpe.done
+		cd := cdata.NewNode()
+		m1 := MockMetricType{
+			namespace: core.NewNamespace("intel", "mock", "foo"),
+		}
+
+		Convey("So metric should be valid with config", func() {
+			errs := c.validateMetricTypeSubscription(m1, cd)
+			So(errs, ShouldBeNil)
+		})
+		c.Config.Plugins.All.AddItem("name", ctypes.ConfigValueStr{Value: "jane"})
+		Convey("So mock should have name: bob config from defaults", func() {
+			So(c.Config.Plugins.pluginCache["0:mock:1"].Table()["name"], ShouldResemble, ctypes.ConfigValueStr{Value: "bob"})
+		})
+
+		c.Stop()
+		time.Sleep(100 * time.Millisecond)
```

Comment:
Seems they were added for Travis issues and I just copied it from another line...

https://github.com/intelsdi-x/snap/commit/da78c23ebea8a5b183e8086e8f8e8171268dfb16
@geauxvirtual